plugin/advisor/pg/advisor_naming_primary_key_convention.go

```diff
@@ -0,0 +1,211 @@
+package pg
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bytebase/bytebase/plugin/advisor"
+	"github.com/bytebase/bytebase/plugin/advisor/catalog"
+	"github.com/bytebase/bytebase/plugin/parser/ast"
+)
+
+var (
+	_ advisor.Advisor = (*NamingPKConventionAdvisor)(nil)
+	_ ast.Visitor     = (*namingPKConventionChecker)(nil)
+)
+
+func init() {
+	advisor.Register(advisor.Postgres, advisor.PostgreSQLNamingPKConvention, &NamingPKConventionAdvisor{})
+}
+
+// NamingPKConventionAdvisor is the advisor checking for primary key naming convention.
+type NamingPKConventionAdvisor struct {
+}
+
+// Check checks for index naming convention.
+func (*NamingPKConventionAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	stmts, errAdvice := parseStatement(statement)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySQLReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+
+	format, templateList, maxLength, err := advisor.UnmarshalNamingRulePayloadAsTemplate(ctx.Rule.Type, ctx.Rule.Payload)
+	if err != nil {
+		return nil, err
+	}
+
+	checker := &namingPKConventionChecker{
+		level:        level,
+		title:        string(ctx.Rule.Type),
+		format:       format,
+		maxLength:    maxLength,
+		templateList: templateList,
+		database:     ctx.Database,
+	}
+
+	for _, stmtNode := range stmts {
+		ast.Walk(checker, stmtNode)
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    advisor.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+
+	return checker.adviceList, nil
+}
+
+type namingPKConventionChecker struct {
+	adviceList   []advisor.Advice
+	level        advisor.Status
+	title        string
+	format       string
+	maxLength    int
+	templateList []string
+	database     *catalog.Database
+}
+
+// Visit implements ast.Visitor interface.
+func (checker *namingPKConventionChecker) Visit(in ast.Node) ast.Visitor {
+	indexDataList := checker.getMetaDataList(in)
+
+	for _, indexData := range indexDataList {
+		regex, err := getTemplateRegexp(checker.format, checker.templateList, indexData.metaData)
+		if err != nil {
```

Comment:
Seems the regex error shouldn't let us skip the maximum length check? We have the same issues in other naming rules

Reply:
Aha, nice advice, let me fix them all in the next PR.